### PR TITLE
Update the vertical styling: styleguide page

### PR DESCRIPTION
**Pull request by sofiyareverse**
https://www.pivotaltracker.com/n/projects/2167210/stories/178778213

### CODE REVIEW

File: app/javascript/packs/application.js [hunk 1]
```diff
@@ -42,9 +42,13 @@ import '@fortawesome/fontawesome-free/scss/solid.scss'
...
 import '../src/dist/css/upside-vendors.css'
 import '../src/dist/js/upside.js'
+import '../src/dist/css/tabler.css'
+import '../src/dist/css/upside-override.css'
+
+import '../src/dist/css/tabler-vendors.css'
+import '../src/dist/css/demo.css'
 
 // TODO: enable when tabler migration is ready
 // import '../src/dist/css/tabler.min.css'
```
Comment (Alexey1100) [inline app/javascript/packs/application.js:54]: We can remove this now

File: app/javascript/packs/application.js [hunk 2]
```diff
@@ -42,9 +42,13 @@ import '@fortawesome/fontawesome-free/scss/solid.scss'
 import 'bootstrap/dist/js/bootstrap.bundle'
 
 import '../src/application.scss'
-import '../src/dist/css/upside.css'
 import '../src/dist/css/upside-vendors.css'
```
Comment (Alexey1100) [inline app/javascript/packs/application.js:45]: And this one

File: app/views/layouts/_header_new_styling.html.erb [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    </h1>
+    <div class="collapse navbar-collapse" id="navbar-menu">
+      <ul class="navbar-nav pt-lg-3">
+        <li class="nav-item">
+          <a class="nav-link" href="/projects/mine">
+            <span class="nav-link-icon d-md-none d-lg-inline-block"><!-- Download SVG icon from http://tabler-icons.io/i/home -->
+              <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-layout-grid" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><rect x="4" y="4" width="6" height="6" rx="1"></rect><rect x="14" y="4" width="6" height="6" rx="1"></rect><rect x="4" y="14" width="6" height="6" rx="1"></rect><rect x="14" y="14" width="6" height="6" rx="1"></rect></svg>
+            </span>
+            <span class="nav-link-title">
+              Projects
```
Comment (Alexey1100) [inline app/views/layouts/_header_new_styling.html.erb:19]: Looks like we are missing some code here which indicates current page.

File: app/views/layouts/_header_new_styling.html.erb [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <h1 class="navbar-brand navbar-brand-autodark">
+      <a href=".">
+        <img src="Logo-Header.svg" width="110" height="32" alt="Tabler" class="navbar-brand-image">
+      </a>
+    </h1>
+    <div class="collapse navbar-collapse" id="navbar-menu">
+      <ul class="navbar-nav pt-lg-3">
+        <li class="nav-item">
```
Comment (Alexey1100) [inline app/views/layouts/_header_new_styling.html.erb:13]: We should add `active` class here, when current page matches the nav bar item url.

**Review by Alexey1100 — CHANGES_REQUESTED** [review]

File: app/javascript/packs/application.js (hunk 1 above)
Comment (sofiyareverse) [reply app/javascript/packs/application.js:54]: But why?

Comment (Alexey1100) [reply app/javascript/packs/application.js:50]: Tabler migration is ready :)

File: app/javascript/controllers/vertical-navbar_controller.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+export default class extends Controller {
+  initialize() {
+    var Handbar = document.querySelector('.main-side-handbar');
+    var MainSide = document.querySelector('#main-side');
+    var Content = document.querySelector('.content');
+    var DropdownMenu = document.querySelector('.dropdown-menu');
+    var Dropdown = document.querySelector('.dropdown');
+    var DropdownBtn = document.querySelector('.dropdown-btn');
+
+    Handbar.addEventListener('click', (event) => {
```
Comment (Alexey1100) [inline app/javascript/controllers/vertical-navbar_controller.js:12]: Minor issue, but could you implement this with stimulus approach (using targets and actions).
Would be easier to maintain and test :)

Ref: https://stimulus.hotwired.dev/handbook/building-something-real

File: app/javascript/packs/application.js [hunk 6]
```diff
@@ -42,9 +42,9 @@ import '@fortawesome/fontawesome-free/scss/solid.scss'
...
 import '../src/dist/js/upside.js'
+import 'packs/new_styling'
```
Comment (Alexey1100) [inline app/javascript/packs/application.js:47]: Should be in `../src/new_styling`

**Comment by Alexey1100** [comment]
Good job :)
A couple of minor styling issues, which should be easy to fix:

1. Dropdown icon is missing:
Screenshot 2021-07-16 at 19 23 07

2. Mouse hover highlight is missing for first-level menu (present in organisation menu):
Screenshot 2021-07-16 at 19 19 25

3. Wallets icon is not centered with other icons:
Screenshot 2021-07-16 at 19 24 29

4. Icons are not the same color as font:
Screenshot 2021-07-16 at 19 25 18
Screenshot 2021-07-16 at 19 32 24


5. Mobile menu has some broken margins:
Screenshot 2021-07-16 at 19 26 25
Screenshot 2021-07-16 at 19 26 32

**Review by Alexey1100 — CHANGES_REQUESTED** [review]
Some UI issues:

1. Organization menu is missing:
Screenshot 2021-07-20 at 22 01 33

2. Close button is not aligned properly, Logo is not centered
Screenshot 2021-07-20 at 22 01 33

**Comment by Alexey1100** [comment]
Good!

I've noticed one more little issue.
The elements are highlighted correctly, but links actually work only if you click on text or icon, so it's a bit confusing.

https://user-images.githubusercontent.com/10757562/126396441-189daba8-ac2f-45b8-95da-d49ef5d31725.mov

**Comment by Alexey1100** [comment]
And I think we should add a feature spec, which will open the page and verify:

1) All links are present and work correctly for every case (unauthorised user / authorised user / authorised comakery admin) both in collapsed and full state.
2) Collapsed state is preserved in local storage between visits.

**Comment by sofiyareverse** [comment]
> And I think we should add a feature spec, which will open the page and verify:
> 
> 1. All links are present and work correctly for every case (unauthenticated user / authenticated user / authenticated comakery admin) both in collapsed and full state.
>    Pls also note and test the recent Tasks menu visibility updates for whitelabel – [Project tasks visibility #2227](https://github.com/CoMakery/comakery-server/pull/2227) and that the Organization menu currently should not be visible in whitelabel.
> 2. Collapsed state is preserved in local storage between visits.

To my mind it will be a good idea only after implementing it to the each page because this page will not exist

**Comment by Alexey1100** [comment]
Let's add it :)
Switching the spec later to any existing page wouldn't be a big deal, and it's really time consuming to verify all of these cases manually during the review.

**Review by zhinfuhnya — APPROVED** [review]

File: app/views/layouts/_header_new_styling.html.erb [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+                    </svg>
+                  </span>
+                </span>
+                <span class="nav-link-title">
+                  Tasks
```
Comment (Alexey1100) [inline app/views/layouts/_header_new_styling.html.erb:69]: Pls integrate Tasks visibility changes from #2227

**Review by Alexey1100 — CHANGES_REQUESTED** [review]

File: app/views/layouts/_header_new_styling.html.erb [hunk 8]
```diff
@@ -55,22 +55,24 @@
...
-                    </svg>
+          <% if @whitelabel_mission.present? && @whitelabel_mission.project_awards_visible? %>
```
Comment (Alexey1100) [inline app/views/layouts/_header_new_styling.html.erb:58]: Should display Tasks if `@whitelabel_mission` is not present or `@whitelabel_mission.project_awards_visible?`

Pls refer to current logic :)

**Review by Alexey1100 — CHANGES_REQUESTED** [review]

**Review by Alexey1100 — APPROVED** [review]

**Pull request merged**